### PR TITLE
Run statistical inference part with `evermore`

### PR DESCRIPTION
**Pull request by pfackeldey**
I kind of blindly coded this so far because I first need some instructions in how to run the full code :D 

But I think this is more or less it. We could extend the `model` with `lnN`s or `shape`-changing templates, or add more parameters (free-floating or constrained) - whatever the plan is.

I'm opening this in draft mode until I could update reqs and actually run it.

### CODE REVIEW

**Comment by pfackeldey** [comment]
@MoAly98 this is ready now :+1:

I can produce the same optimizations with evermore as with the pure jax implementation.

In addition:
- Dropped TF dependency, because I need a newer JAX version, and TF and JAX are not compatible at newer versions. I moved all tensorflow imports into the corresponding functions to protect for import failures.
- Added a private fork of `relaxed` with the fix needed for the pure jax stat implementation to the dependencies

File: analysis/diff.py [hunk 1]
```diff
@@ -35,7 +35,8 @@
 # =============================================================================
 from analysis.base import Analysis
 from user.cuts import lumi_mask
-from utils.jax_stats import build_channel_data_scalar, compute_discovery_pvalue
+# from utils.jax_stats import build_channel_data_scalar, compute_discovery_pvalue
+from utils.evm_stats import build_channel_data_scalar, compute_discovery_pvalue, fit_params
```
[the remark this answers is not in the document]
Comment (MoAly98) [reply analysis/diff.py:39]: - Keep both JAX and EVM implementations for cross-checks, but do not add steering code. 
- TF will turn to legacy analysis -- JAX will be recommended way by Google for NNs with API from Flax NNX (https://flax.readthedocs.io/en/latest/). We can move to this.

File: utils/evm_stats.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+    )
+    # NLL
+    nll = total_loss(fitresult.value, static, channels)
+    # bestfit parameters
+    bestfit_params = evm.tree.combine(fitresult.value, static)
+    return (nll, bestfit_params)
+
+
+@eqx.filter_jit  # JIT compile for performance
+def q0_test(
```
[the remark this answers is not in the document]
Comment (MoAly98) [reply utils/evm_stats.py:129]: need to check how the minimser understands the outercontext when fixed-point differentation is being used, and if there is an issue because the outer gradient descent uses optax while the fit minimiser uses optmisitx

File: utils/evm_stats.py [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+    # Calculate the likelihood ratio
+    # q0 = -2 ln [L(μ=0, θ̂̂) / L(μ̂, θ̂)]
+    likelihood_ratio = 2.0 * (two_nll_conditional - two_nll)
+
+    poi_hat = poi_where(bestfit_params).value
+    q0 = jnp.where(poi_hat >= test_poi, likelihood_ratio, 0.0)
+    # p = 1 - Φ(√q₀)
+    p0 = 1.0 - jax.scipy.stats.norm.cdf(jnp.sqrt(q0))
+    # return p0 and bestfit parameters (as pure values, i.e. without extra info like names, constraints, etc)
+    return (p0, evm.tree.pure(bestfit_params))
```
[the remark this answers is not in the document]
Comment (MoAly98) [reply utils/evm_stats.py:154]: We go with the cov matrix directly for now, and toys can be added later as an option

**Comment by MoAly98** [comment]
@pfackeldey ping me once these are resolved and i wil merge this!
